### PR TITLE
[WIP] Network debug docs

### PR DESCRIPTION
**Pull request by gundalow**
##### SUMMARY
WIP

##### ISSUE TYPE
Docs Pull Request

##### COMPONENT NAME
docs/docsite/rst/network_debug_troubleshooting.rst

##### ANSIBLE VERSION
2.4.0

##### ADDITIONAL INFORMATION

**Nicer view of the docs:**
* https://github.com/gundalow/ansible/blob/network-debug-docs/docs/docsite/rst/intro_networking.rst
* https://github.com/gundalow/ansible/blob/network-debug-docs/docs/docsite/rst/network_how_tos.rst
* https://github.com/gundalow/ansible/blob/network-debug-docs/docs/docsite/rst/network_debug_troubleshooting.rst

### CODE REVIEW

**Comment by gundalow** [comment]
Note to self: may need to add notes, or better yet link to existing documentation on:
```
-* provider (and vars)
-* ssh keys
-* command line (``-u someuser -k``)
-* what else?
```

File: docs/docsite/rst/network_debug_troubleshooting.rst [hunk 1]
```diff
@@ -0,0 +1,658 @@
...
+
+.. notes: Future support
+
+   The list of network platforms that support Persistent Connection will grow with each release.
+
+.. notes: Persistent Connections is for `cli` (ssh), not for API transports.
+
+   The Persistent Connection work added in Ansible 2.3 only applies to `cli transport`. It doesn't apply to APIs such as eos's eapi, or nxos's nxapi. Starting with Ansible 2.3, using CLI should be faster in most cases than using the API transport. Using CLI also allows you be benefit from using SSH Keys.
+
+Playbook Structure from 2.1 to 2.3
```
Comment (dharmabumstead) [inline docs/docsite/rst/network_debug_troubleshooting.rst]: This seems to somewhat duplicate the 'Specifying Credentials' section below.

File: docs/docsite/rst/network_debug_troubleshooting.rst [hunk 2]
```diff
@@ -0,0 +1,658 @@
...
+
+   - name: Gather facts
+     - eos_facts:
+         gather_subset: all
+         transport: eapi
+
+For details on how how to pass in authentication details, see `Specifying Credentials`.
+
+
+Specifying Credentials
```
Comment (dharmabumstead) [inline docs/docsite/rst/network_debug_troubleshooting.rst]: Maybe merge some of the 'Playbook Structure' content detail in with this section.

File: docs/docsite/rst/network_debug_troubleshooting.rst [hunk 3]
```diff
@@ -0,0 +1,658 @@
...
+ ios01                      : ok=1    changed=0    unreachable=0    failed=0
+
+
+This removes the requirement to encode any credentials into the Playbook, further simplifying the Playbook.
+
+Or you can use SSH keys..
+
+FIXME Add details here
+
+Ansible Network Roadmap
```
Comment (dharmabumstead) [inline docs/docsite/rst/network_debug_troubleshooting.rst]: Not sure if this content is something that should be included in this document.

**Review by dharmabumstead — CHANGES_REQUESTED** [review]
See comments. Might be a good idea to move some of the content off into separate docs and/or merge with the existing 'intro to networking' content.

File: docs/docsite/rst/network_debug_troubleshooting.rst [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+
+* Connect to ``switch1.example.net`` specified in the inventory file ``inventory``
+* Use the module ``eos_command``
+* Run the command ``?``
+* Connect using the username ``admin``
+* Inform ansible to prompt for the ssh password by specifying ``-k``
+
+If you have SSH keys configured correctly you can drop the ``-k`` parameter
+
+If the above still fails you can combine it the enable_network_logging_, for example::
```
Comment (Qalthos) [inline docs/docsite/rst/network_debug_troubleshooting.rst]: I don't know what `it the` is supposed to mean

File: docs/docsite/rst/network_debug_troubleshooting.rst [hunk 5]
```diff
@@ -0,0 +1,472 @@
...
+
+:Authentication issues:
+  * Not correctly specifying credentials
+  * Remote device (network switch/router) not falling back to other other authentication methods
+  * SSH key issues
+:Timeout issues:
+  * Can occur when trying to pull a large amount of data
+  * May actually be masking a authentication issue
+:Playbook issues:
+  * Use of ``delgate_to``, instead of ``ProxyCommand``
```
Comment (rcarrillocruz) [inline docs/docsite/rst/network_debug_troubleshooting.rst]: s/delgate_to/delegate_to

**Review by rcarrillocruz — CHANGES_REQUESTED** [review]

File: docs/docsite/rst/network_debug_troubleshooting.rst (hunk 4 above)
Comment (gundalow) [reply docs/docsite/rst/network_debug_troubleshooting.rst]: Fixed

**Comment by gundalow** [comment]
Debug guide is live, so I've moved those changes to a fresh PR https://github.com/ansible/ansible/pull/23735

**Pull request closed without merging**